src/qtviewer/ParseIfcFile.cpp

```diff
@@ -1,18 +1,28 @@
 #include "ParseIfcFile.h"
+#include "MessageLogger.h"
 
-#include "../ifcgeom_schema_agnostic/IfcGeomIterator.h"
+#include "osg/Array"
+#include "osg/Geometry"
+#include "osg/Material"
+#include "osg/PrimitiveSet"
+#include "osg/StateAttribute"
+#include "osg/Vec4"
+#include "osg/ref_ptr"
+#include <OpenGL/OpenGL.h>
+#include <cstddef> // size_t
+#include <string>
+
+#include <osg/Vec3>
+#include <vector>
 
 ParseIfcFile::ParseIfcFile() {}
 
 ParseIfcFile::~ParseIfcFile() {}
 
-void ParseIfcFile::outputMsg(const std::string& msg)
-{
-	emit parsingInfo(QString::fromStdString(msg));
-}
-
-void ParseIfcFile::Parse(const std::string& filePath)
-{
+bool ParseIfcFile::Parse(
+    const std::string& filePath,
+    std::vector<osg::ref_ptr<osg::Geometry>>& geometries
+) {
     IfcParse::IfcFile file(filePath);
 
     IfcGeom::IteratorSettings settings;
```

Comment:
```suggestion
    IfcGeom::IteratorSettings settings;
    settings.set(IfcGeom::IteratorSettings::APPLY_DEFAULT_MATERIALS, true);
```

I would recommend this. If the IFC model does not have materials for some elements, ifcopenshell will then generate defaults. It's one less case to deal with below.

Reply:
Thanks, this is useful and indeed helps avoid additional checks and manual application of default material.
Code updated.